services/web/server/src/simcore_service_webserver/login/handlers_registration.py

```diff
@@ -249,10 +251,17 @@ async def register(request: web.Request):
 
 
 class RegisterPhoneBody(InputSchema):
-    email: EmailStr
+    email: LowerCaseEmailStr
     phone: str = Field(
         ..., description="Phone number E.164, needed on the deployments with 2FA"
     )
+    username: str = Field(None, description="User name", hidden=True)
```

Comment:
OK, so here you are uses

Be aware that RegisterPhoneBody is a  model schema **used in the http API** and you are adding a new field! That means, you **are modifying** the OpenAPI specs of the web-server (see  ``api/specs/webserver/scripts/openapi_auth.py``!)

I see that you are using ``hidden=True`` but I wonder how that manifest in the openapi schema

```
class RegisterPhoneBody(InputSchema):
   email: LowerCaseEmailStr
   

  def get_username(self) -> :
     return self.email.split("@")[0]


```

Reply:
In the end I decided to dont capture the username in the original form, and will be used the one taken already from lowercase email.

Reply:
Yes hidden=True is just for dictionary, we would need to introduce private field so it is ignored in the openapi spec, but that would mean we would need to allow additional fields, which is probably not needed just to capture the original username.